python/cudf/cudf/core/udf/strings_lowering.py

```diff
@@ -36,6 +36,7 @@
 
 
 @cuda_lower(len, MaskedType(string_view))
+@cuda_lower(len, MaskedType(udf_string))
```

Comment:
Was adding the len overload necessarily in scope for this PR?

Reply:
The other thread has more context as to why this is necessary, but this PR actually is adding _all_ the overloads for all of the methods of `string_view` that aren't already attached to `udf_string`. This one just looks weird because it's not an attribute so it's defined off by itself. 

The function of this overload is the following (similar to the others). Here I am dropping the `Masked` view of things for simplicity even though technically we're discussing masked values.

1. Numba sees a call to `len(something)` where the argument being passed (`something`) is a `udf_string`. 
2. All registered typing of `len` are checked to see if one emits a signature when the first argument is a `udf_string`. 
3. The [typing for len](https://github.com/rapidsai/cudf/blob/branch-23.02/python/strings_udf/strings_udf/_typing.py#L129) returns the signature  `len(string_view) -> int32` (this requires the small update in this PR as it's currently returning `len(udf_string) -> int32`
4. Because the signature specifies `string_view` as the argument rather than `udf_string`, the cast from `udf_string` to `string_view` is inserted into the lowering before the rest of the `len` code executes as normal. Note that the fact that this is the effect of the emitted signature expecting the _desired_ type rather than the _input_ type. If we leave it as `len(udf_string) -> int32` it won't insert the cast.